rest/route/project.go

```diff
@@ -433,17 +433,12 @@ func (h *projectIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
 
 	// if owner/repo has changed or we're toggling repo settings off, update scope
 	if h.newProjectRef.Owner != h.originalProject.Owner || h.newProjectRef.Repo != h.originalProject.Repo ||
-		(!h.newProjectRef.UseRepoSettings && h.originalProject.UseRepoSettings) {
+		(!h.newProjectRef.UseRepoSettings() && h.originalProject.UseRepoSettings()) {
 		if err = h.newProjectRef.RemoveFromRepoScope(); err != nil {
 			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "error removing project from old repo scope"))
 		}
 		h.newProjectRef.RepoRefId = "" // if using repo settings, will reassign this in the next block
 	}
-	if h.newProjectRef.UseRepoSettings && h.newProjectRef.RepoRefId == "" {
```

Comment:
So we actually still need this block to exist; users need a way to say "I want to turn on using repo settings" (which is probably why we introduced this field to begin with womp).  I think we could use requestProjectRef to handle this, since APIProjectRef is keeping UseRepoSettings; we'll also need to set this to true (or handle AddToRepoScope) in the above block.

Reply:
ah wait, let me look into using requestProjectRef.

Reply:
Can you please elaborate on "we'll also need to set this to true (or handle AddToRepoScope) in the above block"? It seems to me that if we just swap out h.newProjectRef.UseRepoSettings for h.apiNewProjectRef.UseRepoSettings everything should continue to work. Am I misunderstanding?

Reply:
discussed over slack. done. (I also reset the tests)

Reply:
One last thought: before if `use_repo_settings` was set to false by the user, it was fine if we kept `repo_ref_id` set to whatever, but now we want to explicitly set it to empty. We should handle that here in UpdateProject, i.e. if UseRepoSettings is explicitly false, set RepoRefId to empty. That means I think that UseRepoSettings in APIProjectRef needs to be a boolean pointer.

Reply:
like that?

Reply:
(Apologies for the rebase, the last two commits are new.)